Sources/Plasma/Apps/plPythonPack/main.cpp

```diff
@@ -367,22 +368,37 @@ int main(int argc, char *argv[])
 {
     // Parse arguments
     ST::string packDir = ".";
-    for (int i = 1; i < argc; i++) {
-        ST::string arg = argv[i];
-
-        if (arg.starts_with("?") || arg.starts_with("-") || arg.starts_with("/")) {
-            // Option
-            ST::string option = arg.to_lower().trim_left("/-");
-            if (arg == "?" || option == "h" || option == "help" || option == "?") {
-                PrintUsage();
-                return 0;
-            } else if (option == "q" || option == "quiet") {
-                out = fopen(NULL_DEVICE, "w");
-            }
-        } else {
-            // Path
-            packDir = arg;
+
+    enum { kArgPath, kArgQuiet, kArgHelp1, kArgHelp2 };
+    const plCmdArgDef cmdLineArgs[] = {
+        { kCmdArgOptional | kCmdTypeString, "path",  kArgPath},
+        { kCmdArgFlagged  | kCmdTypeBool,   "quiet", kArgQuiet},
+        { kCmdArgFlagged  | kCmdTypeBool,   "help",  kArgHelp1},
+        { kCmdArgFlagged  | kCmdTypeBool,   "?",     kArgHelp2},
+    };
+
+    std::vector<ST::string> args;
+    args.reserve(argc);
+    for (size_t i = 0; i < argc; i++) {
+        args.emplace_back(ST::string::from_utf8(argv[i]));
+    }
+
+    plCmdParser cmdParser(cmdLineArgs, arrsize(cmdLineArgs));
+    if (cmdParser.Parse(args)) {
+        if (cmdParser.GetBool(kArgHelp1) || cmdParser.GetBool(kArgHelp2)) {
+            PrintUsage();
+            return 0;
         }
+
+        if (cmdParser.GetBool(kArgQuiet))
+            out = fopen(NULL_DEVICE, "w");
```

Comment:
This appears to be leaked at program exit. Not sure how much we care about that though?

Reply:
Amateur mistake.  Who wrote this code?

Reply:
I have added code to `fclose` what I have `fopen`ed.

Reply:
@zrax, do you have an opinion on this matter?

Reply:
I don't think it's super important to `fclose()` a handle to `/dev/null`...  But it's fine do to so, and your implementation has some symmetry to it.  If you want to get really fancy, you could automatically close it with a `std::unique_ptr<FILE, decltype(&fclose)>`, but then you either have to fclose `stdout` (should be safe since you only do it after the application exits) or scope your unique_ptr differently from the `out` handle.

tl;dr:  It looks fine the way you currently implemented it ;)